src/policy/file/v1.rs

```diff
@@ -0,0 +1,346 @@
+        }
+    }
+}
+
+//----------- SignerDenialSpec -------------------------------------------------
+
+/// Spec for generating denial-of-existence records.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields, tag = "type")]
+pub enum SignerDenialSpec {
+    /// Generate NSEC records.
+    NSec,
+
+    /// Generate NSEC3 records.
+    NSec3 {
+        /// Whether and how to enable NSEC3 Opt-Out.
+        opt_out: Nsec3OptOutSpec,
+    },
+}
+
+//--- Conversion
+
+impl SignerDenialSpec {
+    /// Parse from this specification.
+    pub fn parse(self) -> SignerDenialPolicy {
+        match self {
+            SignerDenialSpec::NSec => SignerDenialPolicy::NSec,
+            SignerDenialSpec::NSec3 { opt_out } => SignerDenialPolicy::NSec3 {
+                opt_out: opt_out.parse(),
+            },
+        }
+    }
+
+    /// Build into this specification.
+    pub fn build(policy: &SignerDenialPolicy) -> Self {
+        match *policy {
+            SignerDenialPolicy::NSec => SignerDenialSpec::NSec,
+            SignerDenialPolicy::NSec3 { opt_out } => SignerDenialSpec::NSec3 {
+                opt_out: Nsec3OptOutSpec::build(opt_out),
+            },
+        }
+    }
+}
+
+//--- Default
+
+impl Default for SignerDenialSpec {
+    fn default() -> Self {
+        Self::NSec3 {
+            opt_out: Nsec3OptOutSpec::Disabled,
+        }
+    }
+}
+
+//----------- Nsec3OptOutSpec --------------------------------------------------
+
+/// Spec for the NSEC3 Opt-Out mechanism.
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields, tag = "type")]
+pub enum Nsec3OptOutSpec {
+    /// Do not enable Opt-Out.
+    #[default]
+    Disabled,
+
+    /// Only set the Opt-Out flag.
+    FlagOnly,
+
+    /// Enable Opt-Out and omit the corresponding NSEC3 records.
+    Enabled,
+}
+
+//--- Conversion
+
+impl Nsec3OptOutSpec {
+    /// Parse from this specification.
+    pub fn parse(self) -> Nsec3OptOutPolicy {
+        match self {
+            Nsec3OptOutSpec::Disabled => Nsec3OptOutPolicy::Disabled,
+            Nsec3OptOutSpec::FlagOnly => Nsec3OptOutPolicy::FlagOnly,
+            Nsec3OptOutSpec::Enabled => Nsec3OptOutPolicy::Enabled,
+        }
+    }
+
+    /// Build into this specification.
+    pub fn build(policy: Nsec3OptOutPolicy) -> Self {
+        match policy {
+            Nsec3OptOutPolicy::Disabled => Nsec3OptOutSpec::Disabled,
+            Nsec3OptOutPolicy::FlagOnly => Nsec3OptOutSpec::FlagOnly,
+            Nsec3OptOutPolicy::Enabled => Nsec3OptOutSpec::Enabled,
+        }
+    }
+}
+
+//----------- ReviewSpec -------------------------------------------------------
+
+/// Policy for reviewing loaded/signed zones.
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields, default)]
+pub struct ReviewSpec {
+    /// Whether review is required.
```

Comment:
Required is a bit vague to me. Does it mean the hook won't be invoked, or will be invoked but the result will be ignored?